### PR TITLE
Rename package + Publish on Release

### PR DESCRIPTION
**Pull request by zhiquanyeo**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          registry-url: "https://registry.npmjs.org"
+      # Run yarn install to install project packages
+      - run: npm install
+      # yarn build to build the project
+      - run: npm run build
+      # publish the files to NPM
+      - run: npm publish --access public
+      # for publishing, npm need authorization. We add the NPM token to the environment which will take care of authorization to publish to the package
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (AustinShalit) [inline .github/workflows/main.yml]: End of line at end of file please.
```suggestion
          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}

```

File: package.json [hunk 2]
```diff
@@ -1,12 +1,15 @@
 {
-  "name": "node-wpilib-ws",
+  "name": "@wpilib/node-wpilib-ws",
```
Comment (AustinShalit) [inline package.json:2]: Do we like the name? This is probably the only opportunity we will get to rename it haha

Comment (zhiquanyeo) [reply package.json:2]: True! I'm fine with the name... especially since I don't have any other good ideas lol

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: NPM Publish
+
+# This action will trigger on every release get created
+on:
+  release:
+    types: [created]
```
Comment (AustinShalit) [inline .github/workflows/main.yml:6]: Should this be on tag or on release?

Comment (zhiquanyeo) [reply .github/workflows/main.yml:6]: I was thinking release, just because it seems like a more "final" step when creating a new version

**Comment by zhiquanyeo** [comment]
@AustinShalit if this looks fine to you I'll merge it in and make the corresponding changes for the other 2 repos

**Review by AustinShalit — APPROVED** [review]
Looks good. The only thing I can think of is that we should be running the npm install and build steps on every commit/pr but we can do that in a different pr.

**Pull request merged**